examples/ImageRecon/AtlasNet/eval.py

```diff
@@ -0,0 +1,144 @@
+# You may obtain a copy of the License at
+
+#     http://www.apache.org/licenses/LICENSE-2.0
+
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import os
+import argparse
+import numpy as np
+import torch
+from torch.utils.data import DataLoader
+from tqdm import tqdm
+from easydict import EasyDict
+
+
+from PIL import Image
+
+import kaolin as kal
+from kaolin.datasets import shapenet
+
+from kaolin.models.AtlasNet import AtlasNet
+from kaolin.models.AtlasNet import Resnet18 as Encoder
+
+
+from utils import merge_mesh, preprocess
+
+parser = argparse.ArgumentParser()
+parser.add_argument('--shapenet-root', type=str, help='Root directory of the ShapeNet dataset.')
+parser.add_argument('--shapenet-images-root', type=str, help='Root directory of the ShapeNet Rendering dataset.')
+parser.add_argument('--cache-dir', type=str, default='cache', help='Path to write intermediate representation to.')
+parser.add_argument('--expid', type=str, default='AtlasNet_1', help='Unique experiment identifier.')
+parser.add_argument('--device', type=str, default='cuda', help='Device to use')
+parser.add_argument('--categories', type=str, nargs='+', default=['chair'], help='list of object classes to use')
+parser.add_argument('--no-vis', action='store_true', help='Turn off visualization of each model while evaluating')
+parser.add_argument('--f-score', action='store_true', help='compute F-score')
+parser.add_argument('--batch-size', type=int, default=1, help='Batch size.')
+parser.add_argument('--logdir', type=str, default='log', help='Directory where log data was saved to.')
+args = parser.parse_args()
+
+
+# Data
+points_set_valid = shapenet.ShapeNet_Points(root=args.shapenet_root, cache_dir=args.cache_dir, categories=args.categories,
+                                            train=False, split=.7, num_points=5000)
+images_set_valid = shapenet.ShapeNet_Images(root=args.shapenet_images_root, categories=args.categories,
+                                            train=False, split=.7, views=1, transform=preprocess)
+meshes_set_valid = shapenet.ShapeNet_Meshes(root=args.shapenet_root, categories=args.categories,
+                                            train=False, split=.7)
+valid_set = shapenet.ShapeNet_Combination([points_set_valid, images_set_valid, meshes_set_valid])
+
+dataloader_val = DataLoader(valid_set, batch_size=args.batch_size, shuffle=False,
+                            num_workers=0)
+
+# Model
+opt = EasyDict({
+    "device": args.device,
+    "number_points": 2048,
+    "number_points_eval": 10000,
+    "nb_primitives": 25,
+    "remove_all_batchNorms": False,
+    "template_type": "SQUARE", #Can also be SPHERE
```

Comment:
eval.py:68:31: E261 at least two spaces before inline comment
eval.py:68:32: E262 inline comment should start with '# '